packages/squiggle-lang/src/utility/SDate.ts

```diff
@@ -0,0 +1,110 @@
+import { REOther } from "../errors/messages.js";
+import { SDuration, durationUnits } from "./SDuration.js";
+import { Ok, result, Err } from "./result.js";
+import * as Result from "./result.js";
+
+function isValid(date: Date): boolean {
+  return date instanceof Date && isFinite(date.getTime());
+}
+
+function makeWithYearInt(y: number): result<SDate, string> {
+  if (y < 100) {
+    return Result.Err("Year must be over 100");
+  } else if (y > 200000) {
+    return Result.Err("Year must be less than 200000");
+  } else {
+    return Ok(new SDate(new Date(y, 0)));
+  }
+}
+
+//This is our own internal date class, which is a wrapper around the built-in Date class. It's used by the interpreter, but meant to act like a simple date library.
+export class SDate {
+  constructor(private value: Date) {
```

Comment:
```suggestion
  private constructor(private value: Date) {
```

Same suggestion for `SDuration`. (You'll need to replace one `new SDuration` call in this file with `SDuration.fromMs`)

This is a minor coding style comment, mostly, but I think it's a good practice to make constructors private and expose them only through static methods.

Rationale:
- constructors should be as fast as possible; sometimes you can prove that the arguments are correct and you just need an object without doing any validation
- OTOH, it's dangerous to expose unsafe constructors publicly

This pattern is used a lot in `SymbolicDist`, btw.